### DISCUSSION TITLE
Can struct fuse_session be transferred between two unrelated process ?

### DISCUSSION THREAD

**Discussion by Janos-kz** [libfuse/libfuse#1074] [Q&A] [answered]
Using libfuse3.16.2 lowlevel_api mode
I have got two process A and B
A:
```
fuse_session_new
fuse_session_mount
fuse_session_loop_mt

SIGHUP action: 
fuse_session_exit stop the mt loop
sendmsg to B (send se->fd to B)
```

B:
```
wait on recvmsg
get session fd
fuse_session_new
fuse_session_set_fd (se->fd = fd)
fuse_session_set_mountpoint(se->mountpoint = mountpoint)
fuse_session_set_got_init(se->got_init = 1)
fuse_session_loop_mt
```

Although program B has restarted the loop as imagined, any operation related to the file system will become Input/output error. 

Is there a way for B to perfectly inherit A's fuse_session and block requests on/dev/fuse during runtime ?

**Comment by trapexit** [comment]
libfuse is not designed for that, no. It would have to be modified.

"blocking requests" would simply be not reading or writing. As for transferring the /dev/fuse fd... off hand I don't know if the kernel is checking the identity of the process.

**Reply by bsbernd** [reply, marked as answer]
Kernel is not checking the task struct / process identity.

**Reply by Janos-kz** [reply]
The error is: fuse: writing device: Invalid argument, and I found that the error was finally located at fuse_lowlevel.c : fuse_send_msg:writev(ch ? ch->fd : se->fd, iov, count), the se->fd seems that has expired.
